rendermodules/mesh_lens_correction/LensPointMatches.py

```diff
@@ -263,22 +190,22 @@ def match_image_pairs(render, tilespecs, match_collection, downsample_scale, nfe
     args['match_collection'] = match_collection
     args['nfeature_limit'] = nfeature_limit
     args['matchMax'] = matchMax
-
-    pool = multiprocessing.Pool(ncpus)
     
-    fargs = []
+    with renderapi.client.WithPool(ncpus) as pool:
+    
+        fargs = []
 
-    if index_list is None:
-        index_list = range(tile_index.shape[0])
+        if index_list is None:
+            index_list = range(tile_index.shape[0])
     
-    for i in index_list:
-        impaths = [t.ip.mipMapLevels[0].imageUrl.split('file://')[-1]
+        for i in index_list:
+            impaths = [t.ip.mipMapLevels[0].imageUrl.split('file://')[-1]
```

Comment:
This is fine for a raw data stack, but the current ImagePyramid interface does not guarantee an ordering of mipMapLevels (which are in a list).  In addition, stripping the `file://` lead can cause failures. a more stable way to get the path to the file (for now) is `urllib.unquote(urlparse.urlparse(t.ip.mipMapLevels.get(0)['imageUrl']).path)`.

Reply:
so, if I want to guarantee to retrieve the mipmap level 0 from an ip, do i need to parse the paths? there's nothing else in that data structure, right?
```

mipmapLevels |  
-- | --
0 |  
imageUrl | "file:///allen/programs/celltypes/production/wijem/long_term/em_montage_247488_8R_z0_2018_02_14_00_21_30_00_00/20180213163908673_247488_8R_tape070A_01_000000_0_61_34.tif"
1 |  
imageUrl | "file:/allen/programs/celltypes/production/wijem/mipmap/em_montage_247488_8R_z0_2018_02_14_00_21_30_00_00/1/allen/programs/celltypes/production/incoming/wijem/247488_8R_tape070A_01_000000_0/20180213163908673_247488_8R_tape070A_01_000000_0_61_34.tif.tif"
2 |  
imageUrl | "file:/allen/programs/celltypes/production/wijem/mipmap/em_montage_247488_8R_z0_2018_02_14_00_21_30_00_00/2/allen/programs/celltypes/production/incoming/wijem/247488_8R_tape070A_01_000000_0/20180213163908673_247488_8R_tape070A_01_000000_0_61_34.tif.tif"
3 |  
imageUrl | "file:/allen/programs/celltypes/production/wijem/mipmap/em_montage_247488_8R_z0_2018_02_14_00_21_30_00_00/3/allen/programs/celltypes/production/incoming/wijem/247488_8R_tape070A_01_000000_0/20180213163908673_247488_8R_tape070A_01_000000_0_61_34.tif.tif"
4 |  
imageUrl | "file:/allen/programs/celltypes/production/wijem/mipmap/em_montage_247488_8R_z0_2018_02_14_00_21_30_00_00/4/allen/programs/celltypes/production/incoming/wijem/247488_8R_tape070A_01_000000_0/20180213163908673_247488_8R_tape070A_01_000000_0_61_34.tif.tif"
5 |  
imageUrl | "file:/allen/programs/celltypes/production/wijem/mipmap/em_montage_247488_8R_z0_2018_02_14_00_21_30_00_00/5/allen/programs/celltypes/production/incoming/wijem/247488_8R_tape070A_01_000000_0/20180213163908673_247488_8R_tape070A_01_000000_0_61_34.tif.tif"
6 |  
imageUrl | "file:/allen/programs/celltypes/production/wijem/mipmap/em_montage_247488_8R_z0_2018_02_14_00_21_30_00_00/6/allen/programs/celltypes/production/incoming/wijem/247488_8R_tape070A_01_000000_0/20180213163908673_247488_8R_tape070A_01_000000_0_61_34.tif.tif"
```

Reply:
I'm not sure what you mean by parse the paths in this context -- the url decoding libraries help by, e.g. changing %20 to a space for the filesystem. They also handle less common uris such as file:/path/to/file.

Reply:
Ah, I had a typo in my command which has been edited.  the imagepyramid get method will search mipMapLevels by level.

Reply:
i was referring to `... the current ImagePyramid interface does not guarantee an ordering of mipMapLevels (which are in a list)...`
where I think you were referring to `t.ip.mipMapLevels[0]`
so, my question is, if I can't rely on the first element of the list being mipmap level 0, what's a good strategy?

Reply:
got it, ok.

Reply:
replacing that line with 
impaths = [urllib.unquote(urlparse.urlparse(str(t.ip.mipMapLevels[0].imageUrl)).path)
                    for t in tilespecs[tile_index[i]]]
should work with proper import statements for urllib and urlparse.